stdlib/src/collections/string.mojo

```diff
@@ -2296,6 +2305,17 @@ struct String(
         var result = String(buffer)
         return result^
 
+    fn reserve(inout self, new_capacity: Int):
+        """Reserves the requested capacity.
+
+        If the current capacity is greater or equal, this is a no-op.
+        Otherwise, the storage is reallocated and the date is moved.
+
+        Args:
+            new_capacity: The new capacity.
+        """
```

Comment:
nit:
```suggestion
        Args:
            new_capacity: The new capacity.

        Notes:
            If the current capacity is greater or equal, this is a no-op.
            Otherwise, the storage is reallocated and the data is moved.
        """
```